src/pip/_internal/req/req_install.py

```diff
@@ -130,6 +130,18 @@ def __init__(self, req, comes_from, source_dir=None, editable=False,
         self.isolated = isolated
         self.build_env = NoOpBuildEnvironment()
 
+        # pyproject.toml handling
+        self._pyproject_toml_loaded = False
+        self._pyproject_requires = None
+        self._pyproject_backend = None
+
+        # Are we using PEP 517 for this requirement?
+        # This can be None until load_pyproject_toml has been called,
```

Comment:
It feels a little weird to me though really this is probably the best way to treat the 3-state situation here.

Reply:
Agreed, it's not wonderful, but the alternatives all seemed worse. The idea is that it's like a boolean function argument, with a default value of None (meaning "user didn't specify").

It's tri-state to match the `--[no-]use-pep517` command line option. That will work the same way - `store_true`, `store_false` and a default of None to show "user didn't specify".

I've reworded the comment to clarify a little further.